ecosystem/python/sdk/aptos_sdk/ed25519.py

```diff
@@ -92,6 +94,42 @@ def serialize(self, serializer: Serializer):
         serializer.to_bytes(self.key.encode())
 
 
+class MultiEd25519PublicKey:
+    keys: List[PublicKey]
+    threshold: int
+
+    def __init__(self, keys: List[PublicKey], threshold: int):
+        self.keys = keys
+        self.threshold = threshold
+
+    def __str__(self) -> str:
+        return f"{self.threshold}-of-{len(self.keys)} Multi-Ed25519 public key"
+
+    def auth_key(self) -> bytes:
+        hasher = hashlib.sha3_256()
+        hasher.update(self.to_bytes() + b"\x01")
+        return hasher.digest()
+
+    def to_bytes(self) -> bytes:
+        concatenated_keys = bytes()
+        for key in self.keys:
+            concatenated_keys += key.key.encode()
+        return concatenated_keys + bytes([self.threshold])
+
+    @staticmethod
+    def from_bytes(key: bytes) -> MultiEd25519PublicKey:
```

Comment:
should validate threshold and length

Reply:
@davidiw requested range checks and tests have been added.